--- mod_avmd.c
+++ mod_avmd.c.new
@@ -1483,13 +1483,10 @@
 			flags |= SMBF_READ_REPLACE;
 			direction = "READ_REPLACE";
 	}
-	if ((SWITCH_CALL_DIRECTION_INBOUND == switch_channel_direction(channel)) && (avmd_session->settings.inbound_channnel == 1)) {
-			flags |= SMBF_WRITE_REPLACE;
-			if (!strcmp(direction, "READ_REPLACE")) {
-				direction = "READ_REPLACE | WRITE_REPLACE";
-			} else {
-				direction = "WRITE_REPLACE";
-			}
+	if (SWITCH_CALL_DIRECTION_INBOUND == switch_channel_direction(channel)) {
+			/* DCH: for drachtio-fsmrf */
+			flags |= SMBF_READ_REPLACE;
+			direction = "READ_REPLACE";
 	}
 
 	if (flags == 0) {
